src/stateful-core.ts

```diff
@@ -131,6 +137,10 @@ export default class StatefulCore {
       referrerPageUrl
     });
 
+    const latestId = this.requestManager.getLatestRequestId('universalQuery');
+    if (currentId !== latestId) {
```

Comment:
would this be a little clearer if it read: `if (thisRequestId !== latestRequestId) {`?